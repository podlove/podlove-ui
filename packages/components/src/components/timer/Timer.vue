<script setup lang="ts">
import { computed, ref } from 'vue';
import { toHumanTime } from '@podlove/utils/time'

const props = defineProps({
  time: {
      type: Number,
      default: 0
    },
    remaining: {
      type: Boolean,
      default: false
    }
})

const time = computed(() => toHumanTime(props.time))
</script>

<template>
  <time class="podlove-timer" :datetime="time">{{
    `${remaining ? '-' : ''}${time}`
  }}</time>
</template>


<style lang="postcss">
.podlove-timer {
  color: var(--podlove-component-timer-color, 'currentcolor')
}
</style>
