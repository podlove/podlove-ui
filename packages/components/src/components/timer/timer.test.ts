import { describe, test, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import Timer from './Timer.vue';

describe('Timer', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(Timer);

      expect(wrapper.vm).toBeTruthy();
    });
  });

  describe('props', () => {
    test('should render without props', async () => {
      const wrapper = await mount(Timer);

      expect(wrapper.element).toMatchSnapshot();
    });

    const props = [
      {
        prop: 'color',
        value: '#fff'
      },
      {
        prop: 'time',
        value: 360
      },
      {
        prop: 'remaining',
        value: true
      }
    ];

    props.forEach(({ prop, value }) => {
      test(`should render with prop ${prop}`, async () => {
        const wrapper = await mount(Timer, {
          props: {
            [prop]: value
          }
        });

        expect(wrapper.element).toMatchSnapshot();
      });
    });
  });
});
