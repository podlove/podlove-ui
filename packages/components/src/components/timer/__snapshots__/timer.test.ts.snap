// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Timer > props > should render with prop color 1`] = `
<time
  color="#fff"
  datetime="00:00"
>
  00:00
</time>
`;

exports[`Timer > props > should render with prop remaining 1`] = `
<time
  datetime="00:00"
>
  -00:00
</time>
`;

exports[`Timer > props > should render with prop time 1`] = `
<time
  datetime="00:00"
>
  00:00
</time>
`;

exports[`Timer > props > should render without props 1`] = `
<time
  datetime="00:00"
>
  00:00
</time>
`;
